kedro/framework/cli/starters.py

```diff
@@ -543,64 +542,51 @@ def _get_extra_context(  # noqa: PLR0913
             None in case the flag wasn't used
 
     Returns:
-        the prompts_required dictionary, with all the redundant information removed.
+        Config dictionary, passed the necessary processing, with default values if needed.
     """
     if config_path:
-        extra_context = _fetch_config_from_file(config_path)
-        _validate_config_file_against_prompts(extra_context, prompts_required)
-        _validate_config_file_inputs(extra_context, starter_alias)
+        extra_context = _fetch_validate_parse_config_from_file(
+            config_path, prompts_required, starter_alias
+        )
     else:
-        extra_context = _fetch_config_from_user_prompts(
+        extra_context = _fetch_validate_parse_config_from_user_prompts(
             prompts_required, cookiecutter_context
         )
 
-    # Format
-    extra_context.setdefault("kedro_version", version)
-
-    converted_tools = _convert_tool_names_to_numbers(selected_tools)
-
-    if converted_tools is not None:
-        extra_context["tools"] = converted_tools
-
+    # Update extra_context, if CLI inputs are available
+    if selected_tools is not None:
+        tools_numbers = _convert_tool_short_names_to_numbers(selected_tools)
+        extra_context["tools"] = _convert_tool_numbers_to_readable_names(tools_numbers)
     if project_name is not None:
         extra_context["project_name"] = project_name
+    if example_pipeline is not None:
+        extra_context["example_pipeline"] = str(_parse_yes_no_to_bool(example_pipeline))
 
-    # Map the selected tools lists to readable name
-    tools_context = extra_context.get("tools")
-    tools = _parse_tools_input(tools_context)
-
-    # Check if no tools selected
-    if not tools:
-        extra_context["tools"] = str(["None"])
-    else:
-        extra_context["tools"] = str([NUMBER_TO_TOOLS_NAME[tool] for tool in tools])
-
-    extra_context["example_pipeline"] = _parse_yes_no_to_bool(
-        example_pipeline
-        if example_pipeline is not None
-        else extra_context.get("example_pipeline", "no")
-    )
+    # set defaults for required fields, will be used mostly for starters
+    extra_context.setdefault("kedro_version", version)
+    extra_context.setdefault("tools", str(["None"]))
+    extra_context.setdefault("example_pipeline", "False")
 
     return extra_context
 
 
-def _convert_tool_names_to_numbers(selected_tools: str | None) -> str | None:
-    """Prepares tools selection from the CLI input to the correct format
+def _convert_tool_short_names_to_numbers(selected_tools: str | None) -> list:
```

Comment:
With the current changes, this function should never be invoked with selected_tools as `None`. We can remove `None` from the argument type hint

Reply:
Looks like in the main code - yes, but it used in tests also:
`result = _convert_tool_short_names_to_numbers(None)`
we can rewrite tests also but but maybe it's not so critical

Reply:
I'd rather rewrite the test, as it's not indicative of the actual execution and was probably introduced for test coverage instead of representing a user story

Reply:
correct, fixed!